RUFAS/biophysical/manure/manure_manager.py

```diff
@@ -49,6 +49,122 @@ def __init__(self) -> None:
         self._create_all_processors(processor_connections_by_name, processor_configs_by_name)
         self._populate_adjacency_matrix(processor_connections_by_name)
 
+        self._validate_adjacency_matrix()
+        self._processing_order = self._traverse_adjacency_matrix()  # noqa
+
+    def _validate_adjacency_matrix(self) -> None:
+        """Validates the generated adjacency matrix."""
+        for origin, destinations in self._adjacency_matrix.items():
+            if destinations[origin] != 0:
+                raise ValueError(f"The diagonal for origin {origin} is not 0.")
+            column_sum = sum(destinations.values())
+            if column_sum not in (0, 1):
+                raise ValueError(f"Sum for {origin} column must be 0 or 1, but got {column_sum}")
+
+    def _merge_separator_rows(self) -> dict[str, dict[str, float]]:
+        """
+        Merge the seperator outputs for traversal purposes.
```

Comment:
```suggestion
        Merge the separator outputs for traversal purposes.
```
(if you keep the docstrings as is)